### PR TITLE
add implementation_note element

### PR DESCRIPTION
**Pull request by dericed**
as discussed in this pr https://github.com/cellar-wg/matroska-specification/pull/272#issue-219728738

### CODE REVIEW

File: specification.markdown [hunk 1]
```diff
@@ -476,6 +476,18 @@ A purpose attribute distinguishes the meaning of the documentation. Values for t
...
+
+### \<implementation_note> Attributes
+
+#### note_type
+
+The note_type attribute references which of the `<element>`'s attributes that the implementation_note is in regards to. The note_type attribute MUST be set to one of the following values (corresponding to that attribute of the parent `<element>`): `minOccurs`, `maxOccurs`, `range`, `length`, or `default`. The `<implementation_note>` SHALL supersede the parent `<element>`'s attribute that is named in the `note_type` attribute. An `<element>` SHALL not have more than one `<implementation_note>` of the same `note_type`.
```
Comment (retokromer) [inline specification.markdown]: `SHALL not` > `SHALL NOT`

Comment (dericed) [reply specification.markdown]: updated

**Comment by robUx4** [comment]
Needs an update after 9782f94338596993747dca4f5c6f87e00b0dbf8b (renaming `xsd:` to `xs:`).

File: specification.markdown [hunk 2]
```diff
@@ -476,6 +476,18 @@ A purpose attribute distinguishes the meaning of the documentation. Values for t
...
+
+### \<implementation_note> Attributes
+
+#### note_type
+
+The note_type attribute references which of the `<element>`'s attributes that the implementation_note is in regards to. The note_type attribute MUST be set to one of the following values (corresponding to that attribute of the parent `<element>`): `minOccurs`, `maxOccurs`, `range`, `length`, or `default`. The `<implementation_note>` SHALL supersede the parent `<element>`'s attribute that is named in the `note_type` attribute. An `<element>` SHALL NOT have more than one `<implementation_note>` of the same `note_type`.
```
Comment (robUx4) [inline specification.markdown]: no 'minver' / 'maxver' ? For DivX we MAY explain why elements not found in Matroska can actually be found in Matroska v2 files.

Comment (dericed) [reply specification.markdown]: Good point. Will add.

Comment (dericed) [reply specification.markdown]: added

**Comment by robUx4** [comment]
Rebased so it can be merged (renamed `<xsd:` to `<xs:`).

It will still conflict with #270, whichever is merged first.

**Comment by dericed** [comment]
The prior rebase undid an earlier xsd-cleaning pr, so I cleaned this up and force pushed it. I recommend that this be considered before https://github.com/cellar-wg/ebml-specification/pull/270 as I think there's more consensus here, whereas the extensions may not be necessary to move forth with work on the Matroska EBML Schema.

**Comment by robUx4** [comment]
LGTM

**Pull request merged**